builders/pallets-precompiles/precompiles/gmp.md

```diff
@@ -0,0 +1,155 @@
+
+You may be unfamiliar with both SCALE encoding and MultiLocations if you are not familiar with the Polkadot ecosystem. [SCALE encoding](https://docs.substrate.io/reference/scale-codec/){target=_blank} is a compact form of encoding that Polkadot uses. The [MultiLocation type](https://wiki.polkadot.network/docs/learn-xcvm){target=_blank} is used to define a relative point in Polkadot, such as a specific account on a specific parachain (Polkadot blockchain).  
+
+Moonbeam’s GMP protocol requires a MultiLocation to represent the destination for liquidity routing, which most likely means an account on some other parachain. Whatever it is, this destination must be expressed as relative to Moonbeam.  
+
+!!! remember
+    MultiLocations being relative is important, because a parachain team may erroneously give you a MultiLocation relative to their own chain, which can be different. Providing an incorrect MultiLocation can result in **loss of funds**!   
+
+Each parachain will have their own methods of interpreting a Multilocation, and you will have to confirm with the project that the Multilocation that you form is correct. That being said, it is most likely that you will be forming a Multilocation with an account.
+
+There are multiple types of accounts that can be included in a MultiLocation, which you must know beforehand when constructing your MultiLocation. The two most common are:
+
+- **AccountKey20** — an account id that is 20-bytes in length, including Ethereum-compatible account ids such as those on Moonbeam
+- **AccountId32** — an account id that is 32-bytes in length, standard in Polkadot and its parachains
+
+In the following MultiLocation templates, replace `INSERT_PARACHAIN_ID` with the parachain ID of the network you wish to send funds to and replace `ADDRESS_HERE` with the address of the account you want to send funds to.  
+
+=== "AccountId32"
+    ```json
+    {
+        "parents": 1,
+        "interior": {
+            "X2": [
+                { "Parachain": INSERT_PARACHAIN_ID },
+                { 
+                    "AccountId32": { 
+                        "network": "Any", 
+                        "id": "ADDRESS_HERE" 
+                    } 
+                }
+            ]
+        }
+    }
+    ```
+=== "AccountKey20"
+    ```json
+    {
+        "parents": 1,
+        "interior": {
+            "X2": [
+                { "Parachain": INSERT_PARACHAIN_ID },
+                { 
+                    "AccountKey20": { 
+                        "network": "Any", 
```

Comment:
network any? so this won't work for xcm v3?

Reply:
Both V2 and V3 are supported. Do you think it's worthwhile to convert the code into V3 @eshaben ?

Reply:
I updated to V3 and tested it out. It works well enough and is simpler, so I think I'll keep it at V3 in both the docs and the example I send to Wormhole. Thanks for pointing this out

Reply:
ah sorry I missed the question til now, but sounds like a good plan 👍